### PR TITLE
fix scenarios where `nil` or non map is provided in data causes a panic

### PR DESCRIPTION
**Pull request by andrewpmartinez**
- fixes edge #1020

### CODE REVIEW

File: controller/persistence/config_store.go [hunk 1]
```diff
@@ -62,6 +63,10 @@ func (entity *Config) SetValues(ctx *boltz.PersistContext) {
 	ctx.SetString(FieldName, entity.Name)
 	ctx.SetString(FieldConfigType, entity.Type)
 	ctx.SetMap(FieldConfigData, entity.Data)
+
+	if entity.Data == nil {
+		ctx.Bucket.SetError(errorz.NewFieldError("data is required", "data", nil))
```
Comment (dovholuknf) [inline controller/persistence/config_store.go:68]: is this accurate? wasn't @gberl002 sending data just that data didn't deserialize properly due to the format? would you consider adding "or data improperly formatted" to this response? (assuming I understand what happened)

**Review by dovholuknf — APPROVED** [review]

**Review by gberl002 — APPROVED** [review]

File: controller/persistence/config_store.go (hunk 1 above)
Comment (andrewpmartinez) [reply controller/persistence/config_store.go:68]: Not at this level.

Comment (dovholuknf) [reply controller/persistence/config_store.go:68]: not at this level 'in the code' - but somewhere else? I just want the consumer to get a good message to help them understand the problem. As long as a consumer gets a "hey you screwed up" (speaking of which is this now a 400) and 'how' that's all I care deeply about.

Comment (andrewpmartinez) [reply controller/persistence/config_store.go:68]: https://github.com/openziti/edge/pull/1021/files#diff-aabaaa26930c9df6015bb4c88a9ceae28fafc4e3431545f3c916202c630648cdR50

**Pull request merged**